src/YagrCore/plugins/plotLines/plotLines.ts

```diff
@@ -54,12 +49,22 @@ export default function plotLinesPlugin(options: PlotLineOptions): PlotLinesPlug
 
         const hook = HOOKS_MAP[drawIndicies] || 'drawClear';
 
+        function getLineId(line: PlotLineConfig, scale?: string): string {
+            if (line.id) {
+                return line.id;
+            }
+            const lineWithoutId = Array.from(plotLines.entries()).find(
+                ([_, l]) => deepIsEqual(l, line) && (!scale || line.scale === scale),
+            )?.[0];
+            return lineWithoutId || `${Math.random()}`;
```

Comment:
can you find more reliable method to generate id?

Reply:
Yep, let me work on it)